### ISSUE TITLE
Build Error in Visual Studio when Overriding `TargetFramework` in Projects while Using `Directory.Build.props`

### ISSUE THREAD

**Issue by konarx** [dotnet/msbuild#10967]
### Issue Description

I'm encountering a build error in Visual Studio when using a `Directory.Build.props` file to set the `<TargetFramework>` for my solution, and then override it in specific multitargeted projects.

**Setup:**

- **`Directory.Build.props` at the solution root:**

  ```xml
  <Project>
    <PropertyGroup>
      <TargetFramework>net8.0</TargetFramework>
      <!-- Other common properties -->
    </PropertyGroup>
  </Project>
  ```

- **Projects in the solution:**

  - Multiple projects inherit from `Directory.Build.props` and target `net8.0`.
  - Three specific projects (`ProjectA`, `ProjectB`, `ProjectC`) must target multiple frameworks. In their individual `.csproj` files, they override the `TargetFramework` with `TargetFrameworks`:

    ```xml
    <Project Sdk="Microsoft.NET.Sdk">
      <PropertyGroup>
        <TargetFrameworks>netstandard2.0;netstandard2.1;net8.0</TargetFrameworks>
        <!-- Other project-specific properties -->
      </PropertyGroup>
    </Project>
    ```

**Issue:**

- **Building with Rider and CI Pipelines:**

  - Using Rider (with build tool at `C:\Program Files\dotnet\sdk\8.0.403\MSBuild.dll`), the solution builds successfully.
  - Our CI pipelines also build the solution without issues.

- **Building with Visual Studio 2022 (Up-to-Date):**

  - The build fails with the following error:

    ```
    Severity    Code    Description    Project          File                                                                         Line    Suppression State
    Error       -       Project '..\ProjectC\ProjectC.csproj' targets 'net8.0'. It cannot be referenced by a project that targets '.NETStandard,Version=v2.0'.    ProjectB (netstandard2.0)    C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets    1890
    ```

- **Workaround:**

  - If I remove the `<TargetFramework>` from `Directory.Build.props` and specify it individually in each `.csproj` file, the solution builds successfully in Visual Studio.

**Additional Information:**

- **Environment:**

  - **Visual Studio Version:** Visual Studio 2022 (Up-to-Date)
  - **.NET SDK Version:** 8.0.403
  - **MSBuild Path in Rider:** `C:\Program Files\dotnet\sdk\8.0.403\MSBuild.dll`

- **What Works:**

  - Building the solution using Rider and our CI pipelines works without issues.
  - Removing the `<TargetFramework>` from `Directory.Build.props` and specifying it in each `.csproj` allows Visual Studio to build successfully.

### Steps to Reproduce

1. **Create a Solution with Multiple Projects:**

   - Some projects will inherit `TargetFramework` from `Directory.Build.props`.
   - Three projects (`ProjectA`, `ProjectB`, `ProjectC`) will override `TargetFramework` with `TargetFrameworks` in their `.csproj` files.

2. **Set Up `Directory.Build.props`:**

   - Place a `Directory.Build.props` file at the solution root with the following content:

     ```xml
     <Project>
       <PropertyGroup>
         <TargetFramework>net8.0</TargetFramework>
         <!-- Other common properties -->
       </PropertyGroup>
     </Project>
     ```

3. **Configure the Multitargeted Projects:**

   - In the `.csproj` files for `ProjectA`, `ProjectB`, and `ProjectC`, include:

     ```xml
     <Project Sdk="Microsoft.NET.Sdk">
       <PropertyGroup>
         <TargetFrameworks>netstandard2.0;netstandard2.1;net8.0</TargetFrameworks>
         <!-- Other project-specific properties -->
       </PropertyGroup>
     </Project>
     ```

4. **Build the Solution in Visual Studio 2022:**

   - Attempt to build the solution.

### Expected Behavior

The solution builds successfully, respecting the overridden `TargetFrameworks` in the specified projects.

### Actual Behavior

Visual Studio reports a build error stating that a project targeting `net8.0` cannot be referenced by a project that targets `.NETStandard,Version=v2.0`.

**Error Details:**

```
Severity    Code    Description    Project          File                                                                         Line    Suppression State
Error       -       Project '..\ProjectC\ProjectC.csproj' targets 'net8.0'. It cannot be referenced by a project that targets '.NETStandard,Version=v2.0'.    ProjectB (netstandard2.0)    C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets    1890

### Ask us questions

- Is this behavior expected in Visual Studio when using `Directory.Build.props` to set `TargetFramework`, and then overriding it in individual projects with `TargetFrameworks`?
- Are there any known issues or workarounds to resolve this build error in Visual Studio?

**Comment by konarx** [comment]
**Update:**

I discovered a workaround for the issue by modifying the `Directory.Build.props` file. Instead of using the singular `<TargetFramework>`, I switched to using the plural `<TargetFrameworks>`, even when targeting only a single framework (`net8.0`). This change allows the individual project files to properly override the target frameworks without causing conflicts in Visual Studio.

**Updated `Directory.Build.props`:**

```xml
<Project>
  <PropertyGroup>
    <!-- Suppress MsbuildTargetFrameworkTagInspection -->
    <TargetFrameworks>net8.0</TargetFrameworks>
    <!-- Other common properties -->
  </PropertyGroup>
  <!-- Additional configurations -->
</Project>
```

**Explanation:**

- By using `<TargetFrameworks>` in `Directory.Build.props`, the multitargeted projects can override this property in their `.csproj` files with their own `<TargetFrameworks>` definitions (e.g., `netstandard2.0;netstandard2.1;net8.0`).
- This change prevents the conflict that was occurring between `<TargetFramework>` and `<TargetFrameworks>`, which Visual Studio seemed unable to resolve correctly.
- The `<!-- Suppress MsbuildTargetFrameworkTagInspection -->` comment is used to suppress any inspection warnings about using `<TargetFrameworks>` with a single target.

**Result:**

- After making this change, the solution builds successfully in Visual Studio, Rider, and our CI pipelines.
- Projects that target only `net8.0` inherit the default `<TargetFrameworks>` from `Directory.Build.props` without issues.
- Multitargeted projects correctly override the property with their specific target frameworks.

**Conclusion:**

- While this workaround resolves the immediate build error, it would be helpful to understand why Visual Studio cannot handle the override from `<TargetFramework>` to `<TargetFrameworks>` when defined in `Directory.Build.props`.
- Any insights or guidance on this behavior would be appreciated.
